tests/utilities/test_common.py

```diff
@@ -0,0 +1,97 @@
+import pandas as pd
+import numpy as np
+from hypothesis import given, settings, strategies as st
+from hypothesis.extra.pandas import column, data_frames, range_indexes
+from mawpy.constants import STAY_LAT, STAY_LONG, STAY
+from mawpy.utilities.common import _mean_ignore_minus_ones, _merge_stays, get_combined_stay, get_stay_groups
+
+# Testing _mean_ignore_minus_one()
+series_strategy = st.lists(st.one_of(st.integers(min_value=-1, max_value=100), st.just(-1)),
+                            min_size=1,
+                            max_size=100).map(pd.Series)
+@given(series=series_strategy)
```

Comment:
`@given(df_by_user=df_strategy)
@settings(max_examples=100, deadline=None)
def test_merge_stays(df_by_user):
    unique_stays = df_by_user[STAY].unique()

    stay_to_update = st.sampled_from(unique_stays).map(int)
    updated_stay = st.sampled_from(unique_stays).map(int)
    group_avgs = df_by_user.groupby(STAY)[[STAY_LAT, STAY_LONG]].mean().reset_index()
    group_avgs_index_to_update = group_avgs.loc[group_avgs[STAY] == stay_to_update].index

    df_by_user_expected = df_by_user.copy()
    group_avgs_expected = group_avgs.copy()

    result_df = _merge_stays(stay_to_update, updated_stay, df_by_user, group_avgs, group_avgs_index_to_update)

    # Exclude -1 and compute the mean for latitude and longitude series
    df_by_user_expected.loc[df_by_user_expected[STAY] == stay_to_update, STAY] = updated_stay
`